livekit-plugins/livekit-plugins-deepgram/livekit/plugins/deepgram/tts.py

```diff
@@ -0,0 +1,413 @@
+                async for word in word_stream:
+                    speak_msg = {"type": "Speak", "text": f"{word.token} "}
+                    await ws.send_str(json.dumps(speak_msg))
+
+                # Always flush after a segment
+                flush_msg = {"type": "Flush"}
+                await ws.send_str(json.dumps(flush_msg))
+
+            # after all segments, close
+            close_msg = {"type": "Close"}
+            closing_ws = True
+            await ws.send_str(json.dumps(close_msg))
+
+        async def recv_task(ws: aiohttp.ClientWebSocketResponse):
+            last_frame: rtc.AudioFrame | None = None
+
+            def _send_last_frame(*, segment_id: str, is_final: bool) -> None:
+                nonlocal last_frame
+                if last_frame is not None:
+                    self._event_ch.send_nowait(
+                        tts.SynthesizedAudio(
+                            request_id=request_id,
+                            segment_id=segment_id,
+                            frame=last_frame,
+                            is_final=is_final,
+                        )
+                    )
+                    last_frame = None
+
+            while True:
+                msg = await ws.receive()
+                if msg.type in (
+                    aiohttp.WSMsgType.CLOSE,
+                    aiohttp.WSMsgType.CLOSED,
+                    aiohttp.WSMsgType.CLOSING,
+                ):
+                    if not closing_ws:
+                        raise Exception(
+                            "Deepgram websocket connection closed unexpectedly"
+                        )
+                    return
+
+                if msg.type == aiohttp.WSMsgType.BINARY:
+                    data = msg.data
+                    for frame in audio_bstream.write(data):
+                        _send_last_frame(segment_id=segment_id, is_final=False)
+                        last_frame = frame
+                elif msg.type == aiohttp.WSMsgType.TEXT:
+                    resp = json.loads(msg.data)
+                    mtype = resp.get("type")
+                    if mtype == "Flushed":
+                        for frame in audio_bstream.flush():
+                            _send_last_frame(segment_id=segment_id, is_final=False)
+                            last_frame = frame
+                        _send_last_frame(segment_id=segment_id, is_final=True)
+                    elif mtype == "Warning":
+                        logger.warning("Deepgram warning: %s", resp.get("warn_msg"))
+                    elif mtype == "Metadata":
+                        pass
+                    else:
+                        logger.debug("Unknown message type: %s", resp)
+
+        async def _connection_timeout():
+            # Deepgram has a 60-minute timeout period for websocket connections
+            await asyncio.sleep(3599)
+            logger.warning("Deepgram TTS connection timed out. reconnecting...")
```

Comment:
Change the log to say that this timeout is expected (like maximum connection time reached..)